client/src/components/ReportsByTask.js

```diff
@@ -53,46 +55,11 @@ const Chart = ({
     if (!data) {
       return []
     }
-    const noTaskMessage = `No ${Settings.fields.task.subLevel.shortLabel}`
-    const noTask = {
-      uuid: "-1",
-      shortName: noTaskMessage,
-      longName: noTaskMessage
-    }
-    const reportsList = data.reportList.list || []
-    if (!reportsList.length) {
-      return []
-    }
-    const simplifiedValues = reportsList.map(d => {
-      return { reportUuid: d.uuid, tasks: d.tasks.map(p => p.uuid) }
-    })
-    let tasks = reportsList.map(d => d.tasks)
-    tasks = [].concat
-      .apply([], tasks)
-      .filter(
-        (item, index, d) =>
-          d.findIndex(t => {
-            return t.uuid === item.uuid
-          }) === index
-      )
-      .sort((a, b) => a.shortName.localeCompare(b.shortName))
-    // add No Task item, in order to relate to reports without Tasks
-    tasks.push(noTask)
-    return tasks.map(d => {
-      const r = {}
-      r.task = d
-      r.reportsCount =
-        d.uuid === noTask.uuid
-          ? simplifiedValues.filter(item => item.tasks.length === 0).length
-          : simplifiedValues.filter(item => item.tasks.indexOf(d.uuid) > -1)
-            .length
-      return r
-    })
+    return getReportsByTasks(data.reportList.list || [])
```

Comment:
Could be just `getReportsByTasks(data.reportList.list)` if the method accepts a null/undefined list.